### DISCUSSION TITLE
RuntimeError: MATLAB is selecting SOFTWARE OPENGL rendering. Morphologist 2021

### DISCUSSION THREAD

**Discussion by SC990** [brainvisa/brainvisa.github.io#185] [Q&A]
Hi,

I am trying to run the Morphologist 2021 pipeline. This has previously run successfully; however it is now failing.

(Please note: I am working on a high performance computer cluster which was recently updated by the institution.)

An example of the errors that are appearing: 

in spm12 - Old normalise: Estimate & Write - generic 1
RuntimeError: MATLAB is selecting SOFTWARE OPENGL rendering
Error after SPM finished : 
Matlab execution failed

The matlab location (version2021b1) is stated in preferences -> executable
The spm12 location is stated in preferences -> spm12_path

I would really appreciate any help in solving the error to get the morphologist pipeline to successfully run

**Comment by denisri** [comment]
Hi,
Well, I don't know why Matlab would use some OpenGL rendering here, as it is only used to run SPM normalisation process to get alignment to a standard orientation... Maybe a problem of Matlab on the cluster you are running on, or a problem of Matlab running inside a BrainVisa container on the cluster. And maybe the warning about OpenGL is not even related to the failure of the processing...
Anyway normally you don't need Matlab at all: the BrainVisa container distribution now includes SPM standalone (matlab compiled) and the corresponding MCR (Matlab Compiler Runtime), so you don't even need to configure it, everything is self-contained. Can you try in this setting please ?
Denis

**Comment by SC990** [comment]
Thank you for your quick reply. 

I am using version 5.0.4  as it is important to us that the dataset is all analysed via the same version.

Is this setting possible in this version, and if so, how can i implement it?

**Comment by denisri** [comment]
Ah you're right, Brainvisa 5.0 did not ship SPM yet, as far as I remember..
However I recommend using the standalone version of SPM12. We currently use one with the Matlab MCR v97.
See some clues about it here: https://github.com/spm/spm-docker
(and a long discussion about it here: https://github.com/brainvisa/casa-distro/issues/268)
